packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Snippets/RequestOptionsSnippets.cs

```diff
@@ -11,8 +13,15 @@ namespace Microsoft.Generator.CSharp.ClientModel.Snippets
 {
     internal static class RequestOptionsSnippets
     {
-        public static ScopedApi<RequestOptions> FromCancellationToken()
-            => Static<RequestOptions>().Invoke("FromCancellationToken", [KnownParameters.CancellationTokenParameter]).As<RequestOptions>();
+        public static ScopedApi<RequestOptions> FromCancellationToken(ValueExpression cancellationToken)
+            => new TernaryConditionalExpression(
+                cancellationToken.Property(nameof(CancellationToken.CanBeCanceled)),
+                New.Instance<RequestOptions>(
+                    arguments: [],
+                    properties: new Dictionary<ValueExpression, ValueExpression>
+                        { { new MemberExpression(null, nameof(RequestOptions.CancellationToken)), cancellationToken } },
+                    useSingleLineForPropertyInitialization: true),
+                    Null).As<RequestOptions>();
```

Comment:
we should use abstraction instead of RequetOptions, Azure plugin should use RequetContext instead.

Reply:
fixed in https://github.com/microsoft/typespec/pull/5148